neqo-common/src/datagram.rs

```diff
@@ -124,15 +127,23 @@ impl<D: AsRef<[u8]>> Debug for Datagram<D> {
 }
 
 impl<'a> Datagram<&'a mut [u8]> {
+    /// # Panics
+    ///
+    /// Panics if the data is empty.
     #[must_use]
     pub fn from_slice(src: SocketAddr, dst: SocketAddr, tos: Tos, d: &'a mut [u8]) -> Self {
+        assert!(!d.is_empty(), "Datagram data cannot be empty");
         Self { src, dst, tos, d }
     }
 }
 
 impl Datagram<Bytes> {
+    /// # Panics
+    ///
+    /// Panics if the data is empty.
     #[must_use]
-    pub const fn from_bytes(src: SocketAddr, dst: SocketAddr, tos: Tos, d: Bytes) -> Self {
+    pub fn from_bytes(src: SocketAddr, dst: SocketAddr, tos: Tos, d: Bytes) -> Self {
+        assert!(!d.is_empty(), "Datagram data cannot be empty");
```

Comment:
```suggestion
        assert_ne!(d.len(), 0, "Datagram data cannot be empty");
```

Because `Bytes::len` is `const fn`.

Reply:
Interesting.  I only checked the docs: https://docs.rs/bytes/latest/bytes/struct.Bytes.html#method.len which suggests that it is const.  But then that also shows that is_empty is const.  Could we maybe get an updated crate?

Reply:
Sounds great!